crates/bevy_ecs/src/query/state.rs

```diff
@@ -594,11 +594,14 @@ impl<Q: WorldQuery, F: WorldQuery> QueryState<Q, F> {
         }
     }
 
-    /// Returns an [`Iterator`] over the query results of a list of [`Entity`]'s.
+    /// Returns an [`Iterator`] over the read-only query items generated from an [`Entity`] list.
```

Comment:
This seems a bit less clear than the previous
```
This can only return immutable data (mutable data will be cast to an immutable form).
```
to me. The new version makes it sound like I can't use this with queries that have mutable data. That's not the case, right?

Reply:
Retracting this. I feel like both versions are somewhat confusing but with the link to the `_mut` variant just below, it's pretty easy to figure out how these work.